### PR TITLE
Ignore impossible disambiguations examples in error hint

### PR DESCRIPTION
**Pull request by laurenceisla**
* Ignores m2m relationships where the target and hint are the same table (self reference)
* Ignores m2m relationships where an extra hint is needed to disambiguate

Resolves part of #2070

### CODE REVIEW

**Comment by wolfgangwalther** [comment]
This only removes the mentioned relationships from the hint for the error message - but does not remove from the actual relationship discovery. I don't really understand the purpose of that at all - can you explain what the idea here is?

**Comment by laurenceisla** [comment]
@wolfgangwalther  Yes, this PR is centered only the `hint` key to make it accurate with the examples it gives as a fix to #2031 (to take into consideration the special self reference cases). It does not limit the result in the `details` key as to let the user know that relationships still exist, even if it cannot be disambiguated e.g. the user may ask themselves why the relationship is not detected even if it exists in the database: the details show them that they exist,

From this perspective maybe another key could  be added in the `details` to let the user know where does the relationship in the hint come from:

```json
{
  "hint": "Try changing 'dependencies' to one of the following: 'dependencies!dependencies_locations_fk'. Find the desired relationship in the 'details' key.",
  "details": [
    {
      "relationship": "test.dependencies[dependencies_locations_fk][dependencies_parent_fk]",
      "embedding": "locations with dependencies",
      "cardinality": "many-to-many",
      "disambiguation": "Cannot disambiguate" // here
    },
    {
      "relationship": "dependencies_locations_fk[id][location_id]",
      "embedding": "locations with dependencies",
      "cardinality": "one-to-many"
      "disambiguation": "dependencies!dependencies_locations_fk" // and here
    }
  ],
  ...
}
```

In summary I took it as a simple fix to #2031 and leaves these comments to other PRs: https://github.com/PostgREST/postgrest/issues/2070#issuecomment-996073047 and https://github.com/PostgREST/postgrest/issues/2070#issuecomment-996274638.

**Comment by wolfgangwalther** [comment]
I think that's the wrong approach. If our current implementation of the query language does not allow to disambiguate a request in a way that a certain relationship can actually be used, then we should just not create that relationship in the first place. Listing something in the output for `300 Multiple Choices` that can't actually be chosen is just wrong - and we should solve that. Tweaking the hint provides little value.

**Comment by steve-chavez** [comment]
@laurenceisla Since #2070 is closed perhaps this one should too?

**Comment by laurenceisla** [comment]
@steve-chavez Yes! There is no need for this PR anymore, as the relationship discovery is now restricted in #2262
